cliquet/__init__.py

```diff
@@ -124,23 +125,31 @@ def load_default_settings(config):
 def handle_api_redirection(config):
     """Add a view which redirects to the current version of the API.
     """
-    # Disable the route prefix passed by the app.
-    route_prefix = config.route_prefix
-    config.route_prefix = None
 
     def _redirect_to_version_view(request):
         raise HTTPTemporaryRedirect(
             '/%s/%s' % (route_prefix, request.matchdict['path']))
 
     # Redirect to the current version of the API if the prefix isn't used.
-    config.add_route(name='redirect_to_version',
-                     pattern='/{path:(?!%s).*}' % route_prefix)
+    # Do not redirect if cliquet.version_prefix_redirect_enabled is set to
+    # False.
+    settings = config.get_settings()
+    version_prefix_redirection_enabled = asbool(
+        settings['cliquet.version_prefix_redirect_enabled'])
+
+    if version_prefix_redirection_enabled:
```

Comment:
I suggest that we do that on top, and exit the function (return) if enabled is False. This will prevent to have a long indented bloc of code